.github/workflows/publish-rustdocs.yml

```diff
@@ -42,13 +53,50 @@ jobs:
           docker run --rm \
             -v "${GITHUB_WORKSPACE}:/workspace" \
             -v "${HOME}/.cargo:/root/.cargo" \
-            -w /workspace \
             -e RUSTDOCFLAGS='-D warnings' \
+            -w /workspace \
             paritytech/ci-unified:bullseye-1.74.0 \
-            bash -c "cargo doc --all-features --no-deps --locked && chown -R $(id -u):$(id -g) ${{ env.DOCS_TARGET }}"
+            bash -c "
+              cargo doc --all-features --no-deps --locked && \
+              mv $DOCS_TARGET/help.html $DOCS_TARGET/index.html && chown -R $(id -u):$(id -g) $DOCS_TARGET"
 
-      - name: Deploy to GitHub Pages
+      - name: Set target folder for non-master branches
+        if: github.ref_name != 'master'
+        run: echo "TARGET_FOLDER=${{ github.ref_name }}" >> $GITHUB_ENV
+
+      - name: Push to gh-pages branch
         uses: JamesIves/github-pages-deploy-action@v4
         with:
-          branch: gh-pages-${{ github.ref_name }}
+          branch: ${{ env.BRANCH }}
           folder: ${{ env.DOCS_TARGET }}
+          target-folder: ${{ env.TARGET_FOLDER }}
```

Comment:
What happens if this is then not specified, i.e., we are on `master`?

Reply:
`target-folder=''` hence will be deployed to default root direcotry